static/js/components/dialogs/CollectionFormDialog.js

```diff
@@ -76,28 +78,67 @@ class CollectionFormDialog extends React.Component {
     dispatch(uiActions.setAdminLists(event.target.value));
   };
 
+  validate = () => {
+    const { collectionForm } = this.props;
+    const validation = validateCollectionForm(this.getPayload(collectionForm));
+    this.handleError(validation);
+  };
+
+  getPayload = (collectionForm: CollectionFormState) => {
+    return {
+      title: collectionForm.title,
+      description: collectionForm.description,
+      view_lists: this.calculateListPermissionValue(collectionForm.viewChoice, collectionForm.viewLists),
+      admin_lists: this.calculateListPermissionValue(collectionForm.adminChoice, collectionForm.adminLists),
+    };
+  }
+
   submitForm = async () => {
     const {
       dispatch,
       history,
       collectionUi: { isNew },
-      collectionForm,
+      collectionForm
     } = this.props;
 
-    const payload = {
-      title: collectionForm.title,
-      description: collectionForm.description,
-      view_lists: this.calculateListPermissionValue(collectionForm.viewChoice, collectionForm.viewLists),
-      admin_lists: this.calculateListPermissionValue(collectionForm.adminChoice, collectionForm.adminLists)
-    };
+    const payload = this.getPayload(collectionForm);
 
+    let collection, error;
```

Comment:
It seems like `error` isn't needed here, the error handling could be handled in the `catch` clause. Can you rewrite this like this:

    try {
      if (isNew) {
        collection = await dispatch(actions.collectionsList.post(payload));
        history.push(makeCollectionUrl(collection.key));
      } else {
        await dispatch(actions.collections.patch(collectionForm.key, payload));
      }
      this.onClose();
    } catch (e) {
      this.handleError(e);
    }

Reply:
OK